README.md

```diff
@@ -112,38 +135,1191 @@ sssd::provider::krb5 {'kerberos':
+
+
+
+### sssd
+
+##### `domains`:
+  This key is a mandatory array of domains that will be included in your sssd.conf file. Each of these domains can have an instance of each of the provider defined Types, which must reference the domain that they belong to by name. This list cannot be empty, you must choose at least 1 domain for SSSD to manage.
+  * Valid Options: Array[String]
+  * Default Value: N/A
+
+##### `debug_level`:
+  Sets the debug verbosity for the main section of the config file.
+  * Valid Options: String
+  * Default Value: undef,
+
+##### `debug_timestamps`:
+  Enable or disable timestamps in debug file for this section of the config file.
+  * Valid Options: Boolean
+  * Default Value: true,
+
+##### `debug_microseconds`:
+  Enable or disable microseconds in the debug timestamps for this section of the config file.
+  * Valid Options: Boolean
+  * Default Value: false,
+
+##### `description`:
+  A brief description of this section of the config file.
+  * Valid Options: String
+  * Default Value: undef,
+
+##### `config_file_version`:
+  * Valid Options: Integer
+  * Default Value: 2,
+
+##### `services`:
+  A list of the services that SSSD will integrate with. Each entry here corresponds to one section of the config file at sssd.conf. Each of the services that you include here will be managed by the corresponding sssd::service::service_name subclass.
+  * Valid Options: Sssd::Services
+  * Default Value: ['nss','pam','ssh','sudo'],
+
+##### `reconnection_retries`:
+  Number of times services should attempt to reconnect in the event of a Data Provider crash or restart before they give up
+  * Valid Options: Integer
+  * Default Value: 3,
+
+##### `re_expression`:
+  Default regular expression that describes how to parse the string containing user name and domain into these components.
+  * Valid Options: String
+  * Default Value: undef,
+
+##### `full_name_format`:
+  The Default format that describes how to translate a (name, domain) tuple into a fully qualified name.
+  * Valid Options: String
+  * Default Value: undef,
+
+##### `try_inotify`:
+  Determines if inotify should be used to query resolv.conf
+  * Valid Options: Boolean
+  * Default Value: true,
+
+##### `krb5_rcache_dir`:
+  Directory where krb5 files should be cached.
+  * Valid Options: String
+  * Default Value: undef,
+
+##### `user`:
+  * Valid Options: String
+  * Default Value: undef,
+
+##### `Default_domain_suffix`:
+  Used as the default domain for instances where none is provided
+  * Valid Options: String
+  * Default Value: undef,
+
+##### `override_space`:
+  * Valid Options: String
+  * Default Value: undef,
+
+##### `auditd`:
+  This is a special SIMP level key, which determines automatically if the simp/auditd module is installed on your system. If it is, this module will enable some Defaults to ensure the two modules interact cleanly together
```

Comment:
Other readmes that @RARYates have written discuss simp_options closer to the top of the readme. I think we should standardize that approach.